### PR TITLE
`delete` feature for `key-user-actions-*` and `userActionAndSessionProperties`

### PR DESCRIPTION
**Pull request by jskelin**
#### What this PR does / Why we need it:
Add `delete` functionality for next endpoints:
 * `userActionAndSessionProperties`
 * `key-user-actions-web`
 * `key-user-actions-mobile`

### CODE REVIEW

File: cmd/monaco/integrationtest/account/deploy-download_test.go [hunk 1]
```diff
@@ -94,6 +93,8 @@ func TestIdempotenceOfDeployment(t *testing.T) {
 		assert.Equal(t, deploy1st.Groups[g.ID], deploy2nd.Groups[g.ID])
 	}
 
-	err := monaco("account delete --manifest manifest.yaml --file delete.yaml").withFs(baseFs).run()
-	require.NoError(t, err)
+	t.Cleanup(func() {
+		_, err := test.Monacof("monaco account delete --manifest manifest.yaml --file delete.yaml").WithFs(baseFs).Run()
```
Comment (UnseenWizzard) [inline cmd/monaco/integrationtest/account/deploy-download_test.go:97]: nit: I find it confusing that you're sometimes using this prefixed with 'monaco' and other times without. The fact that the prefix is trimmed anyway is hidden deep in this helper method. I'd either consistently use it with 'monaco' or simply drop the trimming and make it clear from the signature/doc that it takes the command, args and flags that are input to the CLI

File: .editorconfig [hunk 2]
```diff
@@ -11,6 +11,7 @@ insert_final_newline = true
 
 [*.go]
 indent_style = tab
+ij_go_move_all_imports_in_one_declaration = true
```
[the remark this answers is not in the document]
Comment (jskelin) [reply .editorconfig]: moved to another PR

File: pkg/api/api.go [hunk 3]
```diff
@@ -89,3 +89,7 @@ func (a API) Resolve(value string) API {
 	newA.URLPath = strings.ReplaceAll(a.URLPath, "{SCOPE}", value)
 	return newA
 }
+
+func (a API) String() string {
```
[the remark this answers is not in the document]
Comment (jskelin) [reply pkg/api/api.go]: moved to separate PR

File: cmd/monaco/integrationtest/config_rewrite.go [hunk 4]
```diff
@@ -37,7 +37,10 @@ func GenerateTestSuffix(t *testing.T, generalSuffix string) string {
 	}
 
 	timestamp := time.Now().Format("20060102150405")
-	suffix := fmt.Sprintf("%s_%d_%s", timestamp, randomNumber, generalSuffix)
+	suffix := fmt.Sprintf("_%s_%d", timestamp, randomNumber)
+	if generalSuffix != "" {
```
[the remark this answers is not in the document]
Comment (jskelin) [reply cmd/monaco/integrationtest/config_rewrite.go]: moved to separate PR

File: cmd/monaco/integrationtest/cleanup.go [hunk 5]
```diff
@@ -45,6 +45,13 @@ func CleanupIntegrationTest(t *testing.T, fs afero.Fs, manifestPath string, envi
 		_, err := test.Monacof(
 			"monaco generate deletefile %s --file %s --exclude-types builtin:networkzones %s", absManifestPath, deleteFile, env,
 		).WithFs(fs).Run()
+		{
+			if df, err := filepath.Abs(deleteFile); err == nil {
+				if b, err := afero.ReadFile(fs, df); err == nil {
+					fmt.Println(string(b[:]))
```
[the remark this answers is not in the document]
Comment (jskelin) [reply cmd/monaco/integrationtest/cleanup.go]: moved to separate PR

File: pkg/delete/internal/classic/delete.go [hunk 6]
```diff
@@ -21,106 +21,140 @@ import (
...
 
-			var vals []deleteValue
-			vals, err = filterValuesToDelete(logger, scopeEntries, values, theApi.ID)
+// resolveIdentifier get the actual ID from DT and update entries with it
+func resolveIdentifier(ctx context.Context, client dtclient.Client, theAPI *api.API, e *entry) (*api.API, string, error) {
+	var err error
 
-			delValues = append(delValues, vals...)
+	if theAPI.Parent != nil {
+		if _, _, err := resolveIdentifier(ctx, client, theAPI.Parent, e.parent); err != nil && !is404(err) {
```
Comment (arthurpitman) [inline pkg/delete/internal/classic/delete.go:112]: considering that this is calling itself, what exactly does it do? why do we resolve if we dont care about the result?

File: pkg/delete/internal/classic/delete.go [hunk 7]
```diff
@@ -21,106 +21,140 @@ import (
...
 
-	logger.Info("Deleting %d config(s) of type %q...", len(delValues), theApi.ID)
+	id, err := findUniqueID(knownValues, e.identifier)
+	if err != nil {
+		return nil, "", err
+	}
 
-	for _, v := range delValues {
-		vLog := logger.WithFields(field.Coordinate(v.AsCoordinate()), field.F("value", v))
+	e.resolvedID = id
```
Comment (arthurpitman) [inline pkg/delete/internal/classic/delete.go:133]: Do we need to mutate `e`?

File: pkg/delete/internal/classic/delete.go (hunk 6 above)
Comment (arthurpitman) [reply pkg/delete/internal/classic/delete.go:112]: Ok, I see it is because `e` is being updated.

File: pkg/delete/internal/classic/delete.go [hunk 8]
```diff
@@ -21,106 +21,140 @@ import (
...
+	return &a, id, nil
+}
 
-		a := theApi
-		if a.HasParent() {
-			a = a.Resolve(v.DeletePointer.Scope)
-		}
+func findUniqueID(knownValues []dtclient.Value, identifier string) (string, error) {
```
Comment (arthurpitman) [inline pkg/delete/internal/classic/delete.go:137]: How does deletion work? For a given API is it always by name or ID? Or is it really mixed?

File: pkg/delete/internal/classic/delete.go (hunk 6 above)
Comment (arthurpitman) [reply pkg/delete/internal/classic/delete.go:112]: I think probably the complexity introduced by calling itself is not worth the saving in code and would simply be explicit about handling the two cases: with a parent and without.

**Review by warber — APPROVED** [review]

**Pull request merged**